File: src/utils/deep3dApi.js
import axios from 'axios';

/**
 * Sends an image file to the Flask backend to generate a 3D avatar.
 * 
 * @param {File} file - The image file selected by the user
 * @param {number|string} userId - The user ID (can be dynamic)
 * @returns {Promise<string>} - Returns the avatar URL from backend
 */
export const sendToDeep3D = async (file, userId) => {
  try {
    const formData = new FormData();
    formData.append('image', file);
    formData.append('user_id', userId);

    const response = await axios.post('/api/create-avatar', formData, {
      headers: {
        'Content-Type': 'multipart/form-data'
      }
    });

    return response.data.avatar_url;
  } catch (error) {
    console.error('Error sending image to Deep3D:', error);
    throw error;
  }
};

File: src/api/utils/deep3d_api.py
import os
import requests

# 🔧 Mock version – use this while developing locally
def send_to_deep3d(filepath):
    """
    Mock function to simulate sending an image to Deep3D.
    """
    filename = os.path.basename(filepath)
    fake_avatar_url = f"https://mocked-deep3d.com/avatars/{filename.split('.')[0]}.glb"
    print(f"[Mock Deep3D] Returning fake URL: {fake_avatar_url}")
    return fake_avatar_url


# 🧪 Real version (template for production)
def send_to_real_deep3d(filepath):
    """
    Real function to send image to Deep3D API and get avatar URL.
    Update this once you have actual API access.
    """
    api_url = "https://api.deep3d.com/generate-avatar"  # replace with real URL
    api_key = os.getenv("DEEP3D_API_KEY")  # load from .env or environment

    with open(filepath, 'rb') as image_file:
        files = {'image': image_file}
        headers = {
            "Authorization": f"Bearer {api_key}",  # optional, depends on the API
        }

        try:
            response = requests.post(api_url, files=files, headers=headers)
            response.raise_for_status()
            data = response.json()
            return data.get("avatar_url")  # depends on the API's response format

        except Exception as e:
            print(f"[Deep3D Error] {e}")
            return None
